### PR TITLE
Replaced some dialogs with DynamicDialog

### PR DESCRIPTION
**Pull request by d2dyno1**
#2881 introduced `DynamicDialog` - a flexible dialog whose content can be adjusted dynamically from code. This PR replaces simple, existing dialogs with DynamicDialog. These new dynamically created dialogs work just as good but with less code, less files and with more consistent MVVM design.

### CODE REVIEW

**Comment by d2dyno1** [comment]
Ready to be reviewed

**Comment by yaichenbaum** [comment]
@d2dyno1 Can you post before and after screenshots of the dialogs?

**Comment by d2dyno1** [comment]
After:
image
image
image
image

@yair100

**Comment by d2dyno1** [comment]
@yair100 this PR focuses on codebase quality and utilizes consistent MVVM design

**Comment by tsvietOK** [comment]
@d2dyno1 What are the advantages of this DynamicDialog?

**Comment by d2dyno1** [comment]
DynamicDialog can be implemented to replace any other dialog. DynamicDialog aims to remove additional files and gets rid of duplicated code. It's also desined with MVVM design as opposed to existing dialogs. Furthermore, since we have the control of the base [DynamicDialog] any changes made to it will be reflected with other dialogs.
DynamicDialog was originally implemented with #2881 and used throughout Bundles' classes. This PR further improves it and implements it in other places.

@tsvietOK

**Comment by tsvietOK** [comment]
I am not sure about this "improvement". As example, some UI elements generating inside DynamicDialogFactory.cs. It is better to add them via xaml, as it is now.
Before:
image
image
After: size of the dialog changes
image
image
 In case of more complicated Dialog structure (a lot of buttons, data checks, etc.), it will be pretty hard to add it via DynamicDialog. @yair100 what do you think ?

**Comment by yaichenbaum** [comment]
@tsvietOK Complex dialogs can still be custom, but for most dialogs, this is probably a better solution. However, it looks like some more work is needed to make sure the primary buttons use the accent color on the new dialogs before this can be merged.

**Comment by d2dyno1** [comment]
@yair100 All instances of DynamicDialog already use accent color for primary button
image

(I'll work on fixing the unexpected resize with DynamicDialog(RenameDialog) soon) (as @tsvietOK pointed out)
And yes, DynamicDialog is intended to be used on a smaller scale, complex dialogs should still be created separately

**Comment by d2dyno1** [comment]
@tsvietOK the resize has been resolved -> there's now a fixed size

**Comment by d2dyno1** [comment]
@yair100 ready!

File: Files/ViewModels/Dialogs/DynamicDialogViewModel.cs [hunk 1]
```diff
@@ -219,16 +385,19 @@ public DynamicDialogViewModel()
         {
             // Create default implementation
             TitleText = "DynamicDialog";
-            PrimaryButtonText = "OK";
+            PrimaryButtonText = "Ok";
```
Comment (tsvietOK) [inline Files/ViewModels/Dialogs/DynamicDialogViewModel.cs:388]: I guess it is should be translated.

Comment (d2dyno1) [reply Files/ViewModels/Dialogs/DynamicDialogViewModel.cs:388]: It's a fallback value and should always be assigned when creating the dialog. This value should not be used, and I've only kept it here so the button is visible

File: Files/Views/Pages/PropertiesDetails.xaml.cs [hunk 2]
```diff
@@ -36,23 +37,24 @@ public async Task<bool> SaveChangesAsync()
...
                     return true;
                 }
                 catch
                 {
-                    switch (await dialog.ShowAsync())
+                    await dialog.ShowAsync();
+                    switch (dialog.DynamicResult)
                     {
-                        case ContentDialogResult.Primary:
+                        case ViewModels.Dialogs.DynamicResult.Primary:
```
Comment (tsvietOK) [inline Files/Views/Pages/PropertiesDetails.xaml.cs:51]:
```suggestion
                        case ViewModels.Dialogs.DynamicDialogResult.Primary:
```

File: Files/ViewModels/Dialogs/DynamicDialogViewModel.cs [hunk 3]
```diff
@@ -10,31 +11,59 @@ namespace Files.ViewModels.Dialogs
 {
     public enum DynamicResult
```
Comment (tsvietOK) [inline Files/ViewModels/Dialogs/DynamicDialogViewModel.cs:12]: Move it to Enums folder.

File: Files/ViewModels/Dialogs/DynamicDialogViewModel.cs [hunk 4]
```diff
@@ -10,31 +11,59 @@ namespace Files.ViewModels.Dialogs
...
     {
-        Primary = 0,
-        Secondary = 1,
-        Cancel = 2
+        Primary = 1,
+        Secondary = 2,
+        Cancel = 4,
     }
 
     public enum DynamicButtons
```
Comment (tsvietOK) [inline Files/ViewModels/Dialogs/DynamicDialogViewModel.cs:19]: Move it to Enums folder.

Comment (d2dyno1) [reply Files/ViewModels/Dialogs/DynamicDialogViewModel.cs]: kk, done!

**Comment by d2dyno1** [comment]
@tsvietOK anything left todo? ✅

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**